packages/ui-kit/src/components/TimeRangePicker/TimeRangePicker.types.ts

```diff
@@ -0,0 +1,48 @@
+import type { TimeRangeInput } from '../../helpers'
+import type { DefaultThemes } from '../ThemeProvider'
+
+export type DateRangeOptionsProps = {
+  /** A unique identifier for the date range option, used to distinguish between different options. */
+  uid: string
+
+  /** A human-readable label that describes the date range option. */
+  label?: string
+
+  /** The value of the date range option, which can be either a specific date range or a relative time range. */
+  value?: TimeRangeInput
+}
+
+export interface TimeRangePickerProps extends Omit<React.ComponentPropsWithoutRef<'div'>, 'onChange' | 'defaultValue'> {
+  /**
+   * The initial theme used as a base. It provides a default set of styling
+   * from which customizations can be applied.
+   */
+  baseTheme?: DefaultThemes
+
+  /** If set to true, disables the option within the picker that allows users to select a range extending to the current date. */
+  disableDateUntilNow?: boolean
+
+  /** If set to true, disables the option for users to select a custom date range. */
+  disableCustomRange?: boolean
+
+  /** If set to true, disables the option for selecting relative date ranges, such as "last n days." */
+  disableCustomRelative?: boolean
+
+  /** If set to true, disables all selectable options in the component, effectively making it read-only. */
+  disableOptions?: boolean
+
+  /** If true, the entire component is disabled, and users cannot interact with it. */
+  disabled?: boolean
+
+  /** Sets an initial default value for the picker using a structured object that defines a range of options. */
+  defaultValue?: DateRangeOptionsProps
+
+  /** Controls the current value of the picker through a structured object, enabling controlled component behavior. */
+  value?: DateRangeOptionsProps
+
+  /** A function that allows for customizing the list of quick selection options based on default options provided. */
+  options?: (defaultOptions: DateRangeOptionsProps[]) => DateRangeOptionsProps[] | undefined
```

Comment:
Maybe prefixing this with a verb since it's a function? e.g `getOptions`, `buildOptions`

Reply:
Is there any precedent among React libraries for including/omitting a prefix like "get" when the prop's type is a function?

IMO, `getOptions` seems good. But I'd be fine with just `options` if it feels consistent with other libraries.

Also, if we extended the type like this
```suggestion
  options?: DateRangeOptionsProps[] | (defaultOptions: DateRangeOptionsProps[]) => DateRangeOptionsProps[] | undefined
```

then I'd be fine leaving it `options` since you could just return a constant.

Reply:
This is a component's prop in the first place, not a function or a variable. We can define the prop in different ways and a function is just one of them. Besides, we're not getting the options here, we're setting them. The prop's type shouldn't affect the prop's name, otherwise if we decide later to change it from a function to an array, or to a variable (or even to a mix, as @markandrus suggested) – we will have to rename the prop accordingly, which is not very convenient because it will change the component's API.
As for examples, MUI omits prefixing, e.g. Autocomplete's [filterOptions ](https://mui.com/material-ui/api/autocomplete/).

Reply:
@markandrus I would prefer to avoid introducing such a mix of types, because it leads to more complex logic in the component's code without extending the functionality. We can easily pass a fully custom array with the callback: `options={() => [option1, option2, ...]}`